tests/cpp_validation_tests/test_validation.cpp

```diff
@@ -335,29 +400,17 @@ struct CaseParam {
     }
 };
 
-CalculationFunc calculation_func(CaseParam const& param) {
-    auto const get_options = [&param](CalculationMethod calculation_method, Idx threading) {
-        return MainModel::Options{
-            .calculation_type = calculation_type_mapping.at(param.calculation_type),
-            .calculation_symmetry = param.sym ? CalculationSymmetry::symmetric : CalculationSymmetry::asymmetric,
-            .calculation_method = calculation_method,
-            .optimizer_type = param.tap_changing_strategy == "disabled" ? OptimizerType::no_optimization
-                                                                        : OptimizerType::automatic_tap_adjustment,
-            .optimizer_strategy = optimizer_strategy_mapping.at(param.tap_changing_strategy),
-            .err_tol = 1e-8,
-            .max_iter = 20,
-            .threading = threading,
-            .short_circuit_voltage_scaling = sc_voltage_scaling_mapping.at(param.short_circuit_voltage_scaling)};
-    };
-
-    return [get_options](MainModel& model, CalculationMethod calculation_method, MutableDataset const& dataset,
-                         ConstDataset const& update_dataset, Idx threading) {
-        auto options = get_options(calculation_method, threading);
-        if (options.optimizer_type != OptimizerType::no_optimization) {
-            REQUIRE(options.calculation_type == CalculationType::power_flow);
-        }
-        return model.calculate(options, dataset, update_dataset);
-    };
+Options get_options(CaseParam const& param, Idx threading = -1) {
+    Options options{};
+    options.set_calculation_type(calculation_type_mapping.at(param.calculation_type));
+    options.set_calculation_method(calculation_method_mapping.at(param.calculation_method));
+    options.set_symmetric(param.sym ? 1 : 0);
+    options.set_err_tol(1e-8);
+    options.set_max_iter(20);
```

Comment:
Redundant?

Reply:
I don't think so. If the tolerance is reached before 20 iterations, then it is; however, if the tolerance is not reached, the maximum number of iterations will be the safeguard (specially when creating new tests and they don't seem to converge).

Reply:
See the other comment.

Reply:
Making appropriate changes. Resolving.